### PR TITLE
azure cdn log parsing job

### PR DESCRIPTION
**Pull request by xavierdecoster**
Opening the PR for both code and functional review.
Especially try to shoot holes in it from resilience and recovery point of view.
/cc @maartenba @yishaigalatzer @bhuvak @RanjiniM 

**This job's responsibilities:**
* Decompress the *.gz files from blob storage
* Archive the decompressed *.log files in a separate blob container
* Batch insert the log entries into table storage in chronological reverse order and partitioned by date for improved query-ability (metrics/stats jobs will be using this table data as the single source of truth)
* Delete the *.gz file from blob storage after archiving the decompressed log and batch inserting the log entries

**Failure prevention/recovery mechanisms:**
* Job takes a 60 second lease on the blob being processed, avoiding any other process (or job instance when scaling out) to work against the same blob.
* When unable to acquire a lease, the file is ignored as it is assumed to be leased by another process.
* A background thread will auto-renew the blob lease when the parsing duration is nearing lease time
* Upon successful parsing of the blob, the blob is deleted and can no longer be picked up by another process or job instance (all this time, a lease was maintained).
* If any of the following actions fails for a blob, an infinite lease is created on the blob, the blob is moved to a 'dead-letter' container, the JobError is logged in the blob's metadata, and the origin blob is deleted from the source container:
  * decompressing the blob failed (this may indicate a corrupt *.gz file)
  * batch inserting the log entries into table storage failed (even with the configured retry-policy)
  * archiving the decompressed log file failed
  * deleting the source blob failed ('dead-lettering' will give it another try. if it still fails, the job's are ignoring these anyway as they can't acquire a lease on them)
* Table batch operations are grouped by PartitionKey and do not exceed 100 operations per batch

### CODE REVIEW

**Comment by RanjiniM** [comment]
Should there be some integrity check done to make sure all data is present before deleting the *gz files?

File: src/Stats.ParseAzureCdnLogs/JobEventSource.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+
+namespace Stats.ParseAzureCdnLogs
```
Comment (RanjiniM) [inline src/Stats.ParseAzureCdnLogs/JobEventSource.cs]: The new exe model for the jobs have moved away from using event model for tracing. Our V3 jobs uses event log trace listener and V2 jobs use Azure trace listener automatically in addition to event log tracing

Comment (xavierdecoster) [inline src/Stats.ParseAzureCdnLogs/JobEventSource.cs]: Good catch! Seems I took the wrong example as a base for tracing and some jobs still need conversion to the new model. Will adopt EventLog tracing.

As to the define the new model: only EventLog, or both EventLog and Azure tracing? Rather important to have alerts on this one :)

**Comment by xavierdecoster** [comment]
Will reopen this PR against the dev branch.

**Pull request closed without merging**